### PR TITLE
Fix double call to SaveChanges

### LINKED ISSUE #459: Remove duplicate call to SaveChangesAsync()
##### **Overview of the issue**

SaveChangesAsync() is being callled twice for create/update:
At the entity service/controller
And at  the entity repository method CreateOrUpdateAsync

##### **JHipster Version(s)**

Remove call to SaveChangesAsync() at the entity's repository method CreateOrUpdateAsync at the template file `EntityRepository.cs.ejs`.

##### **Browsers and Operating System**

-   [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by ivanmonteiro**
Fix #459

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by ivanmonteiro** [comment]
@nicolas63 Now the service/controller calls SaveChangesAsync.
Please review this PR

File: generators/entity-server/templates/dotnetcore/test/Project.Test/Controllers/EntityResourceIntTest.cs.ejs [hunk 1]
```diff
@@ -296,6 +296,7 @@ namespace <%= namespace %>.Test.Controllers {
         {
             // Initialize the database
             await _<%= camelCasedEntityClass %>Repository.CreateOrUpdateAsync(_<%= camelCasedEntityClass %>);
+            await _<%= camelCasedEntityClass %>Repository.SaveChangesAsync();
```
Comment (ivanmonteiro) [inline generators/entity-server/templates/dotnetcore/test/Project.Test/Controllers/EntityResourceIntTest.cs.ejs:299]: Some tests added entities to the database and needed to call SaveChangesAsync

File: generators/server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/GenericRepository.cs.ejs [hunk 2]
```diff
@@ -92,31 +92,23 @@ namespace <%= namespace %>.Infrastructure.Data.Repositories
             return true;
         }
 
-        public virtual TEntity Remove(TEntity entity)
-        {
```
Comment (ivanmonteiro) [inline generators/server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/GenericRepository.cs.ejs:96]: methods Remove and Delete did the same thing

File: generators/entity-server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/EntityRepository.cs.ejs [hunk 3]
```diff
@@ -73,8 +73,6 @@ namespace <%= namespace %>.Infrastructure.Data.Repositories
                 <%_ } _%>
             }
 
-            await SaveChangesAsync();
```
Comment (ivanmonteiro) [inline generators/entity-server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/EntityRepository.cs.ejs:76]: removed this call to SaveChangesAsync here

File: generators/server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/GenericRepository.cs.ejs [hunk 4]
```diff
@@ -92,31 +92,23 @@ namespace <%= namespace %>.Infrastructure.Data.Repositories
...
         {
             var entity = await GetOneAsync(id);
-            await DeleteAsync(entity);
+            _dbSet.Remove(entity);
         }
         
         public virtual async Task DeleteAsync(TEntity entity)
         {
-            _dbSet.Remove(entity);
-            await SaveChangesAsync();
```
Comment (ivanmonteiro) [inline generators/server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/GenericRepository.cs.ejs:119]: Also removed call to SaveChangesAsync at DeleteAsync method

File: generators/server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/GenericRepository.cs.ejs (hunk 2 above)
Comment (ivanmonteiro) [reply generators/server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/GenericRepository.cs.ejs:96]: so I deleted the Remove method

File: generators/entity-server/templates/dotnetcore/test/Project.Test/Controllers/EntityResourceIntTest.cs.ejs [hunk 5]
```diff
@@ -313,6 +314,7 @@ namespace <%= namespace %>.Test.Controllers {
...
```
Comment (ivanmonteiro) [inline generators/entity-server/templates/dotnetcore/test/Project.Test/Controllers/EntityResourceIntTest.cs.ejs:317]: same here

File: generators/entity-server/templates/dotnetcore/test/Project.Test/Controllers/EntityResourceIntTest.cs.ejs [hunk 6]
```diff
@@ -337,6 +339,7 @@ namespace <%= namespace %>.Test.Controllers {
...
```
Comment (ivanmonteiro) [inline generators/entity-server/templates/dotnetcore/test/Project.Test/Controllers/EntityResourceIntTest.cs.ejs:342]: same here

**Review by nicolas63 — APPROVED** [review]
Perfect 👍

**Pull request merged**